code/modules/mob/living/simple_animal/hostile/retaliate/hog.dm

```diff
@@ -0,0 +1,73 @@
+/mob/living/simple_animal/hostile/retaliate/hog
+	name = "feral hog"
+	desc = "A huge, mangy looking hog. Aggravating it might not be a good idea."
+	icon_state = "hog"
+	icon_living = "hog"
+	icon_dead = "hog_dead"
+	icon_gib = "brownbear_gib"
+	speed = 0
+	maxHealth = 80
+	health = 80
+	harm_intent_damage = 5
+	turns_per_move = 5
+	melee_damage_lower = 20
+	melee_damage_upper = 25
+	response_help_continuous = "squeezes"
+	response_help_simple = "squeeze"
+	attack_verb_continuous = "rams"
+	attack_verb_simple = "ram"
+	butcher_results = list(/obj/item/food/meat/slab = 5) //i would add pork but i don't want to touch crafting menu cooking
+	attack_sound = 'sound/creatures/hog/hogattack.ogg'
+	deathsound = 'sound/creatures/hog/hogdeath.ogg'
+	obj_damage = 80 //do not underestimate the destructive ability of an angry hog
+	environment_smash = ENVIRONMENT_SMASH_STRUCTURES
+	attack_vis_effect = ATTACK_EFFECT_SMASH
+	ranged_cooldown_time = 10 SECONDS
+	speak_emote = list("oinks")
+	var/territorial = TRUE
+	var/rename = TRUE
+
+/mob/living/simple_animal/hostile/retaliate/hog/Initialize(mapload)
+	. = ..()
+	if(territorial)
+		AddComponent(/datum/component/connect_range, src, list(COMSIG_ATOM_ENTERED = PROC_REF(checkEntered)), 1, FALSE)
+	if(rename)
+		switch (gender)
+			if(MALE)
+				name = "feral boar"
+			if(FEMALE)
+				name = "feral sow"
+
+/mob/living/simple_animal/hostile/retaliate/hog/Aggro()
+	..()
+	hogAlert()
+
+/mob/living/simple_animal/hostile/retaliate/hog/proc/checkEntered(datum/source, atom/movable/arrived)
+	if(arrived == src)
+		return
+	if(!ismob(arrived))
+		return
+	Retaliate()
+
+/mob/living/simple_animal/hostile/retaliate/hog/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+	. = ..()
+	if(prob(15))
```

Comment:
does not check for the alive state of the mob, allowing HOG grunting from the afterlife if the body is moved